### PR TITLE
Docs - Add usage for result summary 

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add usage for result summary.

### CODE REVIEW

File: docs/user-tutorial/result-summary.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+## Introduction
+
+This tool is to generate a readable summary report based on the raw benchmark results of single or multiple machines.
+
+## Usage
+
+1. [Install SuperBench](../getting-started/installation) on the local machine.
+
+2. Prepare the raw data, and rule file under the current path or somewhere on the local machine.
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:15]: `2. Prepare the raw data and rule file on the local machine. `

File: docs/user-tutorial/result-summary.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+3. After installing the Superbnech and the files are ready, you can start to generate the result summary automatically using  `sb result summary` command. The detailed command can be found from [SuperBench CLI](../cli).
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:17]: `3. Generate the result summary automatically using `sb result summary` command. The detailed command can be found from [SuperBench CLI](../cli). `

Comment (cp5555) [reply docs/user-tutorial/result-summary.md:17]: one more space after `using`. Please remove it.

File: docs/user-tutorial/result-summary.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  ```
+  sb result summary --data-file ./results-summary.jsonl --rule-file ./rule.yaml --output-file-format md --output-dir ${output-dir}
+  ```
+
+4. After the command is finished, you can find the output result file named 'results_summary.md' under ${output_dir}.
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:23]: `4. Find the output result file named 'results_summary.md' under ${output_dir}.`

File: docs/user-tutorial/result-summary.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+## Input
+
+The input mainly includes 2 files:
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:27]: remove `mainly`

File: docs/user-tutorial/result-summary.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+ - **raw data**: jsonl file including multiple nodes' results automatically generated by SuperBench runner.
+
+    `Tips`: this file can be found at ${output-dir}/results-summary.jsonl after each successful run.
+
+ - **rule file**: It uses YAML format and defines how to generate the result summary including how to classify the metrics and what statistical methods (P50, mean, etc.) to use.
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:33]: what statistical methods (P50, mean, etc.) is applied.

File: docs/user-tutorial/result-summary.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+### rule file
+
+This section describes how to write rules in **rule file**.
+
+The convention is the same with [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:39]: same as

File: docs/user-tutorial/result-summary.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+ - bool:
+    - `False`(default): if no aggregation.
+    - `True`: aggregate the results of multiple ranks. In detail, the metric names in `metrics' like 'metric:\\d+' will be aggregated and turned into 'metric' for most microbenchmark metrics.
+ - pattern string with regex: aggregate the results using the pattern string, which can be used to match the metric names in `metrics`. In detail, the part of the metric that matches the contents of () in the pattern string will be turned into *, other parts of the metric remain unchanged.
+
+#### `statistics`
+
+The list of statistical functions, which are used for this rule, to get the statistics of results from multiple nodes or multiple ranks.
+
+Statistical functions are supported currently:
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:116]: The following illustrates all statistical functions:

File: docs/user-tutorial/result-summary.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+ - `mean`
+ - 'p{value}': {value} can be 1-99, for example, 'p50', 'p90', etc.
+ - 'min'
+ - 'max'
+ - 'std'
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:121]: Would you please change to alphabetical order?

File: docs/user-tutorial/result-summary.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+The value of this item should be bool or pattern string with regex​:
+
+ - bool:
+    - `False`(default): if no aggregation.
+    - `True`: aggregate the results of multiple ranks. In detail, the metric names in `metrics' like 'metric:\\d+' will be aggregated and turned into 'metric' for most microbenchmark metrics.
+ - pattern string with regex: aggregate the results using the pattern string, which can be used to match the metric names in `metrics`. In detail, the part of the metric that matches the contents of () in the pattern string will be turned into *, other parts of the metric remain unchanged.
+
+#### `statistics`
+
+The list of statistical functions, which are used for this rule, to get the statistics of results from multiple nodes or multiple ranks.
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:114]: `The statistical function is used for one specific rule to get the results statistics from multiple nodes/ranks.`

File: docs/user-tutorial/result-summary.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+      statistics: mean
+      categories: RDMA
+      metrics:
+        - ib-loopback/IB_write_8388608_Avg_\d+
+      aggregate: ib-loopback/IB_write_.*_Avg_(\d+)
+```
+
+This rule file describes the rules used for the result summary.
+
+They are firstly organized by the rule name, and each rule mainly includes several elements:
```
Comment (cp5555) [inline docs/user-tutorial/result-summary.md:90]: `They are organized by the rule name and each rule mainly includes several elements: `

File: docs/user-tutorial/result-summary.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+### rule file
+
+This section describes how to write rules in **rule file**.
+
+The convention is the same as [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
+
+Here is an overview of the rule file structure:
+
+scheme:
+```yaml
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:44]:
```suggestion
```yaml title="Scheme"
```

File: docs/user-tutorial/result-summary.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+          - ${statistic_name}
+      categories: string
+      aggregate: (optional)[bool|string]
+      metrics:
+        - ${benchmark_name}/regex
+        - ${benchmark_name}/regex
+```
+
+example:
+```yaml
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:59]:
```suggestion
```yaml title="Example"
```

File: docs/user-tutorial/result-summary.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+    - `False`(default): if no aggregation.
+    - `True`: aggregate the results of multiple ranks. In detail, the metric names in `metrics` like 'metric:\\d+' will be aggregated and turned into 'metric' for most microbenchmark metrics.
+ - pattern string with regex: aggregate the results using the pattern string, which is used to match the metric names in `metrics`. In detail, the part of the metric that matches the contents of () in the pattern string will be turned into *, other parts of the metric remain unchanged.
+
+#### `statistics`
+
+A list of statistical functions is used for this rule to get the results statistics from multiple nodes/ranks.
+
+The following illustrates all statistical functions:
+ - `count`
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:117]: the indent is incorrect?

File: docs/user-tutorial/result-summary.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+  sb result summary --data-file ./results-summary.jsonl --rule-file ./rule.yaml --output-file-format md --output-dir ${output-dir}
+  ```
+
+4. Find the output result file named 'results_summary.md' under ${output_dir}.
+
+## Input
+
+The input includes 2 files:
+
+ - **raw data**: jsonl file including multiple nodes' results automatically generated by SuperBench runner.
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:29]: indent

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:29]: indent

File: docs/user-tutorial/result-summary.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+
+## Usage
+
+1. [Install SuperBench](../getting-started/installation) on the local machine.
+
+2. Prepare the raw data and rule file on the local machine.
+
+3. Generate the result summary automatically using `sb result summary` command. The detailed command can be found from [SuperBench CLI](../cli).
+
+  ```
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:19]:
```suggestion
  ```bash
```

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:19]:
```suggestion
  ```bash
```

File: docs/user-tutorial/result-summary.md [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+
+The input includes 2 files:
+
+ - **raw data**: jsonl file including multiple nodes' results automatically generated by SuperBench runner.
+
+    `Tips`: this file can be found at ${output-dir}/results-summary.jsonl after each successful run.
+
+ - **rule file**: It uses YAML format and defines how to generate the result summary including how to classify the metrics and what statistical methods (P50, mean, etc.) are applied.
+
+### rule file
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:35]: upper cases

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:35]:
```suggestion
### Rule File
```

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:35]:
```suggestion
### Rule File
```

File: docs/user-tutorial/result-summary.md [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+
+4. Find the output result file named 'results_summary.md' under ${output_dir}.
+
+## Input
+
+The input includes 2 files:
+
+ - **raw data**: jsonl file including multiple nodes' results automatically generated by SuperBench runner.
+
+    `Tips`: this file can be found at ${output-dir}/results-summary.jsonl after each successful run.
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:31]:
```
:::tip Tips
......
:::
```

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:31]:
```
:::tip Tips
......
:::
```

File: docs/user-tutorial/result-summary.md [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+
+ - **rule file**: It uses YAML format and defines how to generate the result summary including how to classify the metrics and what statistical methods (P50, mean, etc.) are applied.
+
+### rule file
+
+This section describes how to write rules in **rule file**.
+
+The convention is the same as [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:39]: relative path

Comment (abuccts) [inline docs/user-tutorial/result-summary.md:39]: relative path

**Review by abuccts — COMMENTED** [review]
could you check locally whether it's rendered correctly?

File: docs/user-tutorial/result-summary.md (hunk 1 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 2 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 3 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 4 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 5 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated to 'what statistical methods (P50, mean, etc.) are applied.'

File: docs/user-tutorial/result-summary.md (hunk 6 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 7 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 8 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: sure

File: docs/user-tutorial/result-summary.md (hunk 9 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 10 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 11 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 12 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 13 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: remove the first space?

File: docs/user-tutorial/result-summary.md (hunk 15 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 16 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 18 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md (hunk 17 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

File: docs/user-tutorial/result-summary.md [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+#### `statistics`
+
+A list of statistical functions is used for this rule to get the results statistics from multiple nodes/ranks.
+
+The following illustrates all statistical functions:
+- `count`
+- `max`
+- `mean`
+- `min`
+- `p{value}`: {value} can be 1-99. For example, 'p50', 'p90', etc.
```
Comment (abuccts) [inline docs/user-tutorial/result-summary.md:123]:
```suggestion
- `p${value}`: ${value} can be 1-99. For example, p50, p90, etc.
```

**Review by abuccts — APPROVED** [review]

File: docs/user-tutorial/result-summary.md (hunk 19 above)
Comment (yukirora) [reply docs/user-tutorial/result-summary.md]: updated

**Review by cp5555 — APPROVED** [review]

**Pull request merged**